### ISSUE TITLE
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

### ISSUE THREAD

**Issue by pseudotensor** [Microsoft/LightGBM#1744]
7949cf51c5456fc599c368191ed619418c8b3ca9

During simple predict on simple model:

```
import _pickle as pickle
import lightgbm
model, X, kwargs = pickle.load(open("lgbmmodel.pkl", "rb"))
model.predict(X, **kwargs)
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jon/.pyenv/versions/3.6.4/lib/python3.6/site-packages/lightgbm_gpu/sklearn.py", line 713, in predict
    return self._le.inverse_transform(class_index)
  File "/home/jon/.pyenv/versions/3.6.4/lib/python3.6/site-packages/sklearn/preprocessing/label.py", line 151, in inverse_transform
    if diff:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

```

**Comment by pseudotensor** [comment]
[lgbmmodel.zip](https://github.com/Microsoft/LightGBM/files/2470139/lgbmmodel.zip)

**Comment by pseudotensor** [comment]
Mis-use.  parameter was 3 classes and data only had 2.  But can error checks be put in to handle this so less confusing?

**Comment by guolinke** [comment]
I think this problem is related to python. @StrikerRUS can you help for this issue ?

**Comment by StrikerRUS** [comment]
Fixed in latest versions of `numpy` (1.15.2) and `scikit-learn` (0.20.0):

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-5722436f5b3e> in <module>()
----> 1 model.predict(X, **kwargs)

C:\Anaconda3\lib\site-packages\lightgbm\sklearn.py in predict(self, X, raw_score, num_iteration, pred_leaf, pred_contrib, **kwargs)
    711         else:
    712             class_index = np.argmax(result, axis=1)
--> 713             return self._le.inverse_transform(class_index)
    714 
    715     predict.__doc__ = LGBMModel.predict.__doc__

C:\Anaconda3\lib\site-packages\sklearn\preprocessing\label.py in inverse_transform(self, y)
    279         if len(diff):
    280             raise ValueError(
--> 281                     "y contains previously unseen labels: %s" % str(diff))
    282         y = np.asarray(y)
    283         return self.classes_[y]

ValueError: y contains previously unseen labels: [ 60  70 295]
```
